src/rpc/xmlrpc.cc

```diff
@@ -607,6 +613,357 @@ XmlRpc::set_size_limit(uint64_t size) {
   xmlrpc_limit_set(XMLRPC_XML_SIZE_LIMIT_ID, size);
 }
 
+bool
+XmlRpc::is_valid() const { return m_env != NULL; }
+
+#elif defined(HAVE_XMLRPC_TINYXML2)
```

Comment:
Put these in separate files, `xmlrpc_xmlrpc_c.cc`, `xmlrpc_tinyxml2.cc`, etc.

Reply:
Done, now it uses the config.h flags to enable/disable code, and resolution of which one to use is more of a link time decision. It wasn't my first choice, I did try to go with a OOP model, but that would've leaked changes into other places, including the existing xmlrpc-c code, which seemed to be worth avoiding.